openwisp_users/tasks.py

```diff
@@ -49,17 +54,31 @@ def password_expiration_email():
         )
         .filter(query)
     )
+    email_counts = 0
     for user in qs.iterator():
-        send_email(
-            subject=_('Your password is about to expire'),
-            body_text=_('Your password is about to expire in 7 days'),
-            body_html=_('You password is about to expire in 7 days.'),
-            recipients=[user.email],
-            extra_context={
-                'call_to_action_url': 'https://{0}{1}'.format(
-                    current_site.domain,
-                    reverse('account_change_password'),
-                ),
-                'call_to_action_text': _('Change password'),
-            },
-        )
+        with translation.override(user.language):
+            send_email(
+                subject=_('Action Required: Password Expiry Notice'),
+                body_text=render_to_string(
+                    'account/email/password_expiration_message.txt',
+                    context={'username': user.username, 'expiry_date': expiry_date},
+                ).strip(),
+                body_html=render_to_string(
+                    'account/email/password_expiration_message.html',
+                    context={'username': user.username, 'expiry_date': expiry_date},
+                ).strip(),
+                recipients=[user.email],
+                extra_context={
+                    'call_to_action_url': 'https://{0}{1}'.format(
+                        current_site.domain,
+                        reverse('account_change_password'),
+                    ),
+                    'call_to_action_text': _('Change password'),
+                },
+            )
+        # Avoid overloading the SMTP server by sending multiple
+        # emails continuously.
+        email_counts += 1
+        if email_counts > 10:
```

Comment:
```suggestion
        if email_counts >= 10:
```
I think the current version of the code adds the sleep every 11 emails, not every 10.